lib/priority_parser.rb

```diff
@@ -0,0 +1,46 @@
+#encoding: UTF-8
+
+class PriorityParser
+  PICKUPS_MAP = {
+      id: 0,
+      date: 1,
+      user_id: 2,
+      user_name: 3,
+      user_phone: 4,
+      user_email: 5,
+      supplier_id: 7,
+      supplier_name: 8,
+      status: 9,
+      sub_suppliers: 20.times.map { |i| { id: i * 2 + 10, name: i * 2 + 11 } }
+  }
+
+  def self.process(pickups)
```

Comment:
General note -
wdyt about error handling? should we fail completely if there is a problem with a specific pickup?
  
we should also consider logging the number of pickups we processed, just to make sure that we got it right.